code/Forms/AnchorLinkFormFactory.php

```diff
@@ -10,7 +10,7 @@ protected function getFormFields($controller, $name, $context)
 
         // Ensure current page is selected
         $pageIDField = $fields->dataFieldByName('PageID');
-        $pageIDField->setValue((int)$context['PageID']);
+        $pageIDField->setValue((int)$context['ItemID']);
```

Comment:
This was changed to be more generic so we can reuse it in other link forms if needed.